lib/widgets/recent_dm_conversations.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+
+import '../model/narrow.dart';
+import '../model/recent_dm_conversations.dart';
+import 'content.dart';
+import 'icons.dart';
+import 'message_list.dart';
+import 'page.dart';
+import 'store.dart';
+import 'text.dart';
+
+class RecentDmConversationsPage extends StatefulWidget {
+  const RecentDmConversationsPage({super.key});
+
+  static Route<void> buildRoute({required BuildContext context}) {
+    return MaterialAccountPageRoute(context: context,
+      builder: (context) => const RecentDmConversationsPage());
+  }
+
+  @override
+  State<RecentDmConversationsPage> createState() => _RecentDmConversationsPageState();
+}
+
+class _RecentDmConversationsPageState extends State<RecentDmConversationsPage> with PerAccountStoreAwareStateMixin<RecentDmConversationsPage> {
+  RecentDmConversationsView? model;
+
+  @override
+  void onNewStore() {
+    model?.removeListener(_modelChanged);
+    model = PerAccountStoreWidget.of(context).recentDmConversationsView
+      ..addListener(_modelChanged);
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in [model].
+      // This method was called because that just changed.
+    });
+  }
+
+  Widget _buildItem(BuildContext context, DmNarrow narrow) {
+    final allRecipientIds = narrow.allRecipientIds;
+    final store = PerAccountStoreWidget.of(context);
+    final selfUser = store.users[store.account.userId]!;
+    final recipientsSansSelf = allRecipientIds
+      .whereNot((id) => id == selfUser.userId)
+      .map((id) => store.users[id]!)
+      .toList();
+
+    final Widget title;
+    final Widget avatar;
+    switch (recipientsSansSelf.length) {
+      case 0: {
+        title = Text(selfUser.fullName);
+        avatar = Avatar(userId: selfUser.userId);
+        break;
+      }
+      case 1: {
+        final otherUser = recipientsSansSelf.single;
```

Comment:
Can switch on `recipientsSansSelf` itself rather than its length, and then this can say
```dart
      case [var otherUser]:
```

(Or combined with my other comment above, it'd say `var otherUserId` and then look that up in `store.users`.)